packages/@aws-cdk/aws-ecs/lib/base/base-cluster.ts

```diff
@@ -0,0 +1,63 @@
+import cloudwatch = require ('@aws-cdk/aws-cloudwatch');
+import ec2 = require('@aws-cdk/aws-ec2');
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from '../ecs.generated';
+
+/**
+ * Basic cluster properties
+ */
+export interface BaseClusterProps {
+  /**
+   * A name for the cluster.
+   *
+   * @default CloudFormation-generated name
+   */
+  clusterName?: string;
+
+  /**
+   * The VPC where your ECS instances will be running or your ENIs will be deployed
+   */
+  vpc: ec2.VpcNetworkRef;
+}
+
+/**
+ * Base class for Ecs and Fargate clusters
+ */
+export abstract class BaseCluster extends cdk.Construct {
```

Comment:
What about `ClusterRef`?

Reply:
We have `IEc2Cluster` and `IFargateCluster`. No need for an abstraction of the base class.